test/e2e/upgrade_test.go

```diff
@@ -31,15 +31,71 @@ const (
 	windowsWorkloadTesterJob = "windows-workload-tester"
 )
 
+// getNodeAffinities generate node affinity for non-BYOH and BYOH nodes if present
+func getNodeAffinities() ([]*v1.Affinity, error) {
+	var affinities []*v1.Affinity
+	if gc.numberOfMachineNodes >= 1 {
+		affinityNoByoh, err := getNodeAffinityForLabel(v1.NodeSelectorOpDoesNotExist, controllers.BYOHLabel)
```

Comment:
I think we should name this variable different for byoh and no byoh scenario.

Reply:
Good catch, missed this one.